packages/paypal-commerce-integration/src/paypal-commerce/paypal-commerce-payment-strategy.spec.ts

```diff
@@ -410,6 +424,165 @@ describe('PayPalCommercePaymentStrategy', () => {
+                    acs_url: null,
+                    payer_auth_request: null,
+                    merchant_data: null,
+                    callback_url: null,
+                },
+                errors: [
+                    {
+                        code: 'invalid_request_error',
+                        message:
+                            'Were experiencing difficulty processing your transaction. Please contact us or try again later.',
+                    },
+                    {
+                        code: 'transaction_rejected',
+                        message: 'Payment was declined. Please try again.',
+                        provider_error: {
+                            code: 'INSTRUMENT_DECLINED',
+                        },
+                    },
+                ],
+            };
+
+            jest.spyOn(paypalCommerceIntegrationService, 'submitPayment').mockImplementation(() => {
+                // eslint-disable-next-line @typescript-eslint/no-throw-literal
+                throw providerError;
+            });
+
+            await strategy.initialize(initializationOptions);
+
+            try {
+                await strategy.execute(payload);
+            } catch (_error: unknown) {
+                expect(paypalCommerceIntegrationService.loadPayPalSdk).toHaveBeenCalled();
+            }
+        });
+
+        it('renders paypal spb if receive INSTRUMENT_DECLINED error and experiment is on', async () => {
+            paymentMethod.initializationData.orderId = '1';
+
+            const payload = {
+                payment: {
+                    methodId: defaultMethodId,
+                },
+            };
+
+            jest.spyOn(
+                paymentIntegrationService.getState(),
+                'getStoreConfigOrThrow',
+            ).mockReturnValue(storeConfigWithFeaturesOn);
+
+            const providerError = {
+                status: 'error',
+                three_ds_result: {
+                    acs_url: null,
+                    payer_auth_request: null,
+                    merchant_data: null,
+                    callback_url: null,
+                },
+                errors: [
+                    {
+                        code: 'invalid_request_error',
+                        message:
+                            'Were experiencing difficulty processing your transaction. Please contact us or try again later.',
+                    },
+                    {
+                        code: 'transaction_rejected',
+                        message: 'Payment was declined. Please try again.',
+                        provider_error: {
+                            code: 'INSTRUMENT_DECLINED',
+                        },
+                    },
+                ],
+            };
+
+            jest.spyOn(paypalCommerceIntegrationService, 'submitPayment').mockImplementation(() => {
+                // eslint-disable-next-line @typescript-eslint/no-throw-literal
```

Comment:
Take a look at this [comment](https://github.com/bigcommerce/checkout-sdk-js/pull/2320/files#r1452452476) about moving rule to appropriate file

Reply:
added